### PR TITLE
Upgrading install script to install & prepare a Postgres + DfE Sign-in.

### PR DESCRIPTION
**Pull request by NyWeb**
### Context
Getting both backend and frontend up and running took me (Rails novice) 30min, so the instructions are good. I ran into issues with Postgres, and DfE sign-in seems to be a requirement.

##### DfE Sign-in
Technically DfE Sign-in is needed in front-end, not in this library, but the commands are run on mcb, hence it being here.

Traditionally a god-user created another god-user. That gives the new user access to all current providers - with the awkward side-effect of an inverse relationship between the number of providers on your profile and the amount of time you've been a dev for Find. Here I humbly guess that adding specific providers directly to your user might be the way to go for future devs.

### Changes proposed in this pull request
Adding a chapter on DfE sign-in, created with @timabell's help. Also added a user-creation command to get past a macOS specific bug.

### Guidance to review
First PR on this project, to give feedback

### Checklist
- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by NyWeb** [comment]
[README update for macOS + DfE Sign-in](https://trello.com/c/GWEssvOl/1786-readme-update-for-macos-dfe-sign-in)

File: README.md [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
 1. Run `bundle install` to install the gem dependencies
 2. Run `bundle exec rails db:setup` to create a development and testing database
+   1. MacOS: If you get an error `FATAL: role "postgres" does not exist` run `createuser -s postgres` in your terminal
```
Comment (defong) [inline README.md]: I think this applies to postgres regard less of OS

@timabell for input

File: README.md [hunk 2]
```diff
@@ -191,3 +192,21 @@ be organised in an appropriate sub-folder there.
   device you need BYOD & 2FA set up.
 * A publish user with your email address with access to the organisation(s) you
   want to modify.
+
+### DfE Sign-in
+
+To use it locally (and in qa) you will need a DfE Sign-in account.
```
Comment (defong) [inline README.md]: maybe reword it to state along the veins of `mandatory requirements` 
ie
to successfully login to system
1.  need to create an account on dfe
2.  assign organisation/provider or via god user

Otherwise you stuck

File: README.md [hunk 3]
```diff
@@ -191,3 +192,21 @@ be organised in an appropriate sub-folder there.
...
+Clicking on `Create an account` asks you to 
+"Contact the approver of your organisation". We're within DfE itself, so
+that's not terribly helpful.
+
+##### How to get approved and get access to test data:
+1. Get a DfE Sign-in admin to invite you (give them this link: 
+https://signin-test-sup-as.azurewebsites.net/users)
+2. Sign up from the email sent from DfE Sign-in
+3. Grant access to providers (here on production): 
+`bundle exec bin/mcb users -E production grant {email} {course_code}`
```
Comment (defong) [inline README.md]: mismatch

bullet point 1 uses tests url
bullet point 3 uses production

File: README.md [hunk 4]
```diff
@@ -191,3 +192,21 @@ be organised in an appropriate sub-folder there.
...
+
+Note that all users are anonymized and synced to `qa` daily,
```
Comment (defong) [inline README.md]: reword it the other way please, all other environments uses anonymized not just `qa` or `localhost`

File: README.md [hunk 5]
```diff
@@ -47,7 +48,7 @@ docker-compose exec web /bin/sh -c "bundle exec rails db:setup"
 
 Then open http://localhost:3001 to see the app.
 
-## Running specs, linter(without auto correct) and annotate models and serializers
+## Running specs, linter (without auto correct) and annotate models and serializers
```
Comment (defong) [inline README.md]: I think its with auto correct

@dankmitchell for input

File: README.md (hunk 1 above)
Comment (NyWeb) [reply README.md]: Ok, MacOS removed :)

File: README.md (hunk 5 above)
Comment (NyWeb) [reply README.md]: It's more legible and correct I believe.

File: README.md (hunk 4 above)
Comment (NyWeb) [reply README.md]: How about "Users not matching `%@digital.education.gov.uk` and `%@education.gov.uk` will be anonymized."?

File: README.md (hunk 3 above)
Comment (NyWeb) [reply README.md]: Added context

File: README.md (hunk 2 above)
Comment (NyWeb) [reply README.md]: Thanks, much clearer now!

File: README.md (hunk 5 above)
Comment (defong) [reply README.md]: opps I meant this bit

linter (without auto correct)
vs
linter (with auto correct)

not the spacing linter(with auto correct)

File: README.md [hunk 6]
```diff
@@ -191,3 +192,19 @@ be organised in an appropriate sub-folder there.
...
+### Mandatory requirements
+
+To successfully log into the system, you will need to:
+1. Create an account on DfE Sign-in
+   1. Get a DfE Sign-in admin to invite you (give them this link: 
+      https://signin-test-sup-as.azurewebsites.net/users)
+   2. Sign up from the email sent from DfE Sign-in
+2. Grant access to some providers: 
+   1. For local: `bundle exec bin/mcb users grant {email} {course_code}`
+   2. For `qa` and `production`: `bundle exec bin/mcb -E production users grant {email} {course_code}`
```
Comment (defong) [inline README.md]: prefer correct ordering

```
2. For `qa` and `production`:
  1. You will need to log into Azure first: `az login`
  2. `bundle exec bin/mcb -E {env} users grant {email} {provider_code}`
```
should be  {provider_code} vs {course_code}

File: README.md [hunk 7]
```diff
@@ -191,3 +192,19 @@ be organised in an appropriate sub-folder there.
...
+      1. You will need to log into Azure first: `az login`
+
+Users not matching
+`%@digital.education.gov.uk` and `%@education.gov.uk`
+will be anonymized.
```
Comment (defong) [inline README.md]:
```suggestion
will be anonymized for non production environment.
```

File: README.md (hunk 6 above)
Comment (NyWeb) [reply README.md]: Apologies, what does `should be {provider_code} vs {course_code}` mean?

Comment (defong) [reply README.md]: didn't check the command...

but grant a user access to a provider I would think it needs provider code vs the course code
https://github.com/DFE-Digital/manage-courses-backend/blob/f517332bb30864e57946da82f940e68feb10e049/lib/mcb/commands/users/grant.rb#L4

so i guess command should be something like, 
`
`bundle exec bin/mcb -E {env} grant {email} {provider_code}`
`

**Comment by NyWeb** [comment]
After a conversation with @dankmitchell and @misaka I renamed and updated `lib/tasks/create_dev_user.rake` to install Postgres and fix the user issue we've had. I also restructured that script quite a bit - hopefully for the better, but I left one switch clause too many perhaps.
